tests/playwright/src/model/pages/create-machine-page.ts

```diff
@@ -0,0 +1,137 @@
+    });
+    this.podmanMachineCPUs = this.podmanMachineConfiguration.getByRole('slider', { name: 'CPU(s)' });
+    this.podmanMachineMemory = this.podmanMachineConfiguration.getByRole('slider', { name: 'Memory' });
+    this.podmanMachineDiskSize = this.podmanMachineConfiguration.getByRole('slider', { name: 'Disk size' });
+    this.rootPriviledgesCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', {
+      name: 'Machine with root privileges',
+    });
+    this.userModeNetworkingCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', {
+      name: 'User mode networking',
+    });
+    this.startNowCheckbox = this.podmanMachineConfiguration.getByRole('checkbox', { name: 'Start the machine now' });
+    this.closeButton = this.page.getByRole('button', { name: 'Close' });
+    this.createMachineButton = this.page.getByRole('button', { name: 'Create' });
+  }
+
+  async createMachine(
+    machineName: string,
+    { isRootful = true, enableUserNet = false, startNow = true, setAsDefault = true },
+  ): Promise<ResourcesPage> {
+    await playExpect(this.podmanMachineConfiguration).toBeVisible();
+    await this.podmanMachineName.fill(machineName);
+
+    await this.ensureCheckboxState(isRootful, this.rootPriviledgesCheckbox);
+    await this.ensureCheckboxState(enableUserNet, this.userModeNetworkingCheckbox);
+    await this.ensureCheckboxState(startNow, this.startNowCheckbox);
+
+    await this.createMachineButton.click();
+    await this.page.waitForTimeout(60_000);
+
+    const successfulCreationMessage = this.page.getByText('Successful operation');
+    const goBackToResourcesButton = this.page.getByRole('button', { name: 'Go back to resources' });
+
+    await delay(5000);
+    await this.handleConnectionDialog(machineName, setAsDefault);
+
+    await playExpect(successfulCreationMessage).toBeVisible({ timeout: 10_000 });
+    await playExpect(goBackToResourcesButton).toBeVisible();
+    await goBackToResourcesButton.click();
+
+    return new ResourcesPage(this.page);
+  }
+
+  async ensureCheckboxState(desiredState: boolean, checkbox: Locator): Promise<void> {
+    if (desiredState !== (await this.isEnabled(checkbox))) {
+      await this.switchCheckbox(checkbox);
+    }
+  }
+
+  async isEnabled(checkbox: Locator): Promise<boolean> {
+    await playExpect(checkbox).toBeVisible();
+    const upperElement = checkbox.locator('..').locator('..');
+    const clickableCheckbox = upperElement.getByText('Enabled');
+    return await clickableCheckbox.isVisible();
+  }
+
+  async switchCheckbox(checkbox: Locator): Promise<void> {
+    await playExpect(checkbox).toBeVisible();
+    const upperElement = checkbox.locator('..').locator('..');
+
+    const wasEnabled = await this.isEnabled(checkbox);
+    let checkText;
+    if (wasEnabled) {
+      checkText = 'Enabled';
+    } else {
+      checkText = 'Disabled';
+    }
+
+    const clickableCheckbox = upperElement.getByText(checkText);
+    await clickableCheckbox.click();
```

Comment:
After performing the switching the new state should be asserted upon to ensure that the value of the checkbox actually changed.